AppCenter/AppCenter/Internals/Channel/MSOneCollectorChannelDelegate.m

```diff
@@ -96,31 +120,67 @@ - (void)channel:(id<MSChannelProtocol>)channel didSetEnabled:(BOOL)isEnabled and
   }
 }
 
-- (void)channel:(id<MSChannelProtocol>)channel didPrepareLog:(id<MSLog>)log withInternalId:(NSString *)__unused internalId {
-  if (![self shouldSendLogToOneCollector:log] ||
-      ![channel conformsToProtocol:@protocol(MSChannelUnitProtocol)]) {
-    return;
-  }
-  id<MSChannelUnitProtocol> channelUnit = (id<MSChannelUnitProtocol>)channel;
-  NSString *groupId = channelUnit.configuration.groupId;
-  id<MSChannelUnitProtocol> oneCollectorChannelUnit = [self.oneCollectorChannels objectForKey:groupId];
-  if (!oneCollectorChannelUnit) {
-    return;
-  }
-  id<MSLogConversion> logConversion = (id<MSLogConversion>)log;
-  NSArray<MSCommonSchemaLog *> *commonSchemaLogs = [logConversion toCommonSchemaLogs];
-  for (MSCommonSchemaLog *commonSchemaLog in commonSchemaLogs) {
-    [oneCollectorChannelUnit enqueueItem:commonSchemaLog];
-  }
-}
+#pragma mark - Helper
 
 - (BOOL)isOneCollectorGroup:(NSString *)groupId {
   return [groupId hasSuffix:kMSOneCollectorGroupIdSuffix];
 }
 
 - (BOOL)shouldSendLogToOneCollector:(id<MSLog>)log {
   NSObject *logObject = (NSObject *)log;
-  return [[log transmissionTargetTokens] count] > 0 && [log conformsToProtocol:@protocol(MSLogConversion)] && ![logObject isKindOfClass:[MSCommonSchemaLog class]];
+  return [[log transmissionTargetTokens] count] > 0 && [log conformsToProtocol:@protocol(MSLogConversion)] &&
+         ![logObject isKindOfClass:[MSCommonSchemaLog class]];
+}
+
+- (BOOL)validateLog:(MSCommonSchemaLog *)log {
+  if (![self validateLogName:log.name]) {
+    return NO;
+  }
+  if (![self validateLogData:log.data]) {
+    return NO;
+  }
+  return YES;
+}
+
+- (BOOL)validateLogName:(NSString *)name {
+
+  // Name mustn't be nil.
+  if (!name.length) {
```

Comment:
Since we have the regex you don't need to check for empty. But that's fine.

Reply:
It's so light to do it in objective c that I decided to go with it ;)

Reply:
I like that it's easy to spot that we're considering the case of empty name - having this in regex makes sense but is a little too obfuscated for my taste.